File: ClientRuntimes/NodeJS/lib/credentials/basicAuthenticationCredentials.js
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information. 

var util = require('util');
var Constants = require('../constants');
var validate = require('../validate');

var HeaderConstants = Constants.HeaderConstants;
var DEFAULT_AUTHORIZATION_SCHEME = 'Basic';

/**
* Creates a new BasicAuthenticationCredentials object.
*
* @constructor
* @param {string} credentials.authorizationScheme    The authorization scheme.
* @param {string} credentials.userName               User name.
* @param {string} credentials.password               Password.
*/
function BasicAuthenticationCredentials(credentials) {
  validate.validateArgs('BasicAuthenticationCredentials', function (v) {
    v.object(credentials, 'credentials');
    v.string(credentials.token, 'credentials.userName');
    v.string(credentials.token, 'credentials.password');
  });

  if (!credentials.authorizationScheme) {
    credentials.authorizationScheme = DEFAULT_AUTHORIZATION_SCHEME;
  }

  this.credentials = credentials;
}

/**
* Signs a request with the Authentication header.
*
* @param {WebResource} The WebResource to be signed.
* @param {function(error)}  callback  The callback function.
* @return {undefined}
*/
BasicAuthenticationCredentials.prototype.signRequest = function (webResource, callback) {
  webResource.headers[HeaderConstants.AUTHORIZATION] = util.format('%s %s:%s', 
  	this.credentials.authorizationScheme, this.credentials.userName, this.credentials.password);

  callback(null);
};

module.exports = BasicAuthenticationCredentials;

File: ClientRuntimes/Java/client-runtime/src/main/java/com/microsoft/rest/credentials/BasicAuthenticationCredentials.java
/**
 *
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 *
 */

package com.microsoft.rest.credentials;

import okhttp3.OkHttpClient;

/**
 * Basic Auth credentials for use with a REST Service Client.
 */
public class BasicAuthenticationCredentials implements ServiceClientCredentials {

    /**
     * Basic auth UserName.
     */
    private String userName;

    /**
     * Basic auth password.
     */
    private String password;

    /**
     * Instantiates a new basic authentication credential.
     *
     * @param userName basic auth user name
     * @param password basic auth password
     */
    public BasicAuthenticationCredentials(String userName, String password) {
        this.userName = userName;
        this.password = password;
    }

    /**
     * Get the user name of the credential.
     *
     * @return the user name
     */
    public String getUserName() {
        return userName;
    }

    /**
     * Get the password of the credential.
     *
     * @return the password
     */
    public String getPassword() {
        return password;
    }

    @Override
    public void applyCredentialsFilter(OkHttpClient.Builder clientBuilder) {
        clientBuilder.interceptors().add(new BasicAuthenticationCredentialsInterceptor(this));
    }
}
